lightningd/options.c

```diff
@@ -191,6 +191,38 @@ static char *opt_add_addr(const char *arg, struct lightningd *ld)
 	return opt_add_addr_withtype(arg, ld, ADDR_LISTEN_AND_ANNOUNCE, true);
 }
 
+static char *opt_alt_subdaemon(const char *arg, struct lightningd *ld)
+{
+	// example arg: "hsmd:rmthsmd"
+	char *argbuf = tal_strdup(ld, arg);
+	char *colon = strchr(argbuf, ':');
+	if (!colon)
+	{
+		tal_free(argbuf);
```

Comment:
We'll free its parent immediately anyway?

Reply:
fixed